src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -201,14 +203,34 @@ protected function propertyIdentifier(\EntityMetadataWrapper $property_wrapper)
     if ($property_wrapper instanceof \EntityDrupalWrapper) {
       // The property wrapper is a reference to another entity get the entity
       // ID.
-      $embedded_identifier = $property_wrapper->getIdentifier();
-    }
-    else {
-      // The property is a regular one, get the value out of it and use it as
-      // the embedded identifier.
-      $embedded_identifier = $this->fieldValue($property_wrapper);
+      $identifier = $property_wrapper->getIdentifier() ?: NULL;
+      $resource = $this->getResource();
+      // TODO: Make sure we still want to support full_view.
```

Comment:
You are right, it may not make sense for JSON API but it does for HAL and SimpleJson